### DISCUSSION TITLE
Add Cron Job configuration support to Monorepos

### DISCUSSION THREAD

**Discussion by chriseagle** [vercel/community#5419] [Idea]
Hello,

Current support for `vercel.json` and per project configuration option setting in a monorepo seems lacking. When deploying using a Monorepo with multiple Vercel projects linked to individual apps in said Monorepo, you can only provide a single `vercel.json` in the root of the repo. When configuring cron jobs, this means that all projects then have the configured cron jobs added.

1. It is currently not possible to configure a per project `vercel.json` file in each project which is deployed
2. It is currently not possible to disable individual Cron jobs from the U.I, only option is to disable Cron entirely on a per project basis. 

This could be solved with an addition to `vercel.json` where specific configuration options like Cron can be matched to project names/ids.

Setting the root dir option to the specific project sub folder, doesn't appear to solve this, as it moves the cwd during build to the project root, which in turn means a standard monorepo setup which runs additional scripts from repo root will not function correctly. 

Would love to find out if another way to configure this is possible with the current set of tools, but to me this feels a bit of a shortcoming currently. 

Thanks

**Comment by AndyBitz** [comment]
Hey @chriseagle, thanks for bringing this up!

I assume your projects require the build command to be executed in the repo's root directory and that doing `cd` at build time won't work?

Depending on your setup and build process, you might be able to use the [Build Output API](https://vercel.com/docs/build-output-api/v3/configuration#crons) directly. In this scenario, you could add something to your build script to modify the `.vercel/output` directory. However, this only works when the build script itself creates the `.vercel/output` directory. In the case of Next.js for example, at least at the moment, it would create that directory only after the build process is done, so it wouldn't be possible.

The `--prebuilt` flag for the Vercel CLI could mitigate this, but that's far from ideal.

We'll look into different solutions to support this.

**Comment by hunterwilhelm** [comment]
I am just starting my journey attacking this problem, I'll keep this post updated as best I can with updates.

### May 20th, 2024

Got 1 cron job working without affecting the other projects in the mono repo.

1. Create a `vercel.json` with the cron info in it not in the root, but in your project subdirectory.
```json
{
  "crons": [{
    "path": "/api/cron",
    "schedule": "0 10 * * *"
  }]
}
```
2. `vercel link --repo` (alpha flag) Choose the project to push to.
3. `vercel build --prod && vercel deploy --prebuilt --prod` (faster deployment for me)

### TODO

- [ ] Figure out a way to switch between `vercel link --repo` quickly to deploy to just one project at a time (ignore build step is slow for me).
